### PR TITLE
Add platform model

### PR DESCRIPTION
**Pull request by d2lam**
- Add platform model 
- Remove the check if scmUrl already exists inside pipeline model's `create`. This will be the plugin's responsibility as discussed with @nkatzman yesterday. Open to suggestion on this.
- Remove unnecessary fields

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -25,30 +25,22 @@ class PipelineModel extends BaseModel {
...
+        const platform = 'generic@1';
+        const pipelineData = hoek.applyToDefaults({
+            createTime,
+            platform,
+            configUrl: config.scmUrl
+        }, config);
+        const pipelineConfig = {
+            table: this.table,
+            params: {
+                id: pipelineId,
```
Comment (nkatzman) [inline lib/pipeline.js]: should just call the variables `id` and `data`

File: lib/platform.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Create a platform
+     * @method create
+     * @param  {Object}   config                Config object to create the platform with
+     * @param  {String}   config.name           The name of the platform
+     * @param  {String}   config.version        The version of the platform
+     * @param  {Function} callback              fn(err)
+     */
+    create(config, callback) {
+        const id = hashr.sha1({
```
Comment (nkatzman) [inline lib/platform.js:25]: We keep track of the of unique keys that should be used for hashing here: https://github.com/screwdriver-cd/data-schema/blob/master/model/platform.js#L97

Should we find a way to generically implement a hashing function for all models?

Comment (d2lam) [inline lib/platform.js:25]: Let's do this in a separate PR

**Comment by nkatzman** [comment]
👍

File: README.md [hunk 3]
```diff
@@ -8,6 +8,78 @@
...
+```
+
+#### Create
+Create a new platform
+```
+create(config, callback)
+```
+
+| Parameter        | Type  | Required | Description |
+| :-------------   | :---- | | :---- | :-------------|
```
Comment (tkyi) [inline README.md]: Extra `|` on this line.

**Comment by tkyi** [comment]
Just need to fix the Create table in the README
👍
